File: app/src/main/java/com/kindleassistant/api/RestApi.java
package com.kindleassistant.api;


import com.kindleassistant.sender.model.entity.PreViewRequest;
import com.kindleassistant.sender.model.entity.PreViewRsp;
import com.kindleassistant.sender.model.entity.SendRequest;

import java.util.List;

import io.reactivex.Observable;
import kotlin.Unit;
import okhttp3.MultipartBody;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

public interface RestApi {

    @POST("send")
    Observable<Unit> send(@Body SendRequest send);

    @POST("preview")
    Observable<PreViewRsp> preview(@Body PreViewRequest send);

    @Multipart
    @POST("upload")
    Call<Void> upload(@Part List<MultipartBody.Part> multipartBody);
}

File: app/src/main/java/com/kindleassistant/api/RestApi.kt
package com.kindleassistant.api


import com.kindleassistant.sender.model.entity.PreViewRequest
import com.kindleassistant.sender.model.entity.PreViewRsp
import com.kindleassistant.sender.model.entity.SendRequest
import io.reactivex.Observable
import okhttp3.MultipartBody
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part

interface RestApi {

    @POST("send")
    fun send(@Body send: SendRequest): Observable<Response<Unit>>

    @POST("preview")
    fun preview(@Body send: PreViewRequest): Observable<PreViewRsp>

    @Multipart
    @POST("upload")
    fun upload(@Part multipartBody: List<MultipartBody.Part>): Observable<Response<Unit>>
}
